packages/dds/merge-tree/src/mergeTree.ts

```diff
@@ -1302,6 +1303,82 @@ export class MergeTree {
 		return undefined;
 	}
 
+	/**
+	 * Finds the nearest reference with ReferenceType.Tile to `startPos` in the direction dictated by `posPrecedesTile`.
+	 * Uses depthFirstNodeWalk in addition to the block-accelerated functionality introduced for findTile. The search
+	 * position will be included in the nodes to walk, so searching on all positions, including the endpoints, can be considered
+	 * inclusive. Any out of bound search positions will return undefined, so in order to search the whole string, a forward
+	 * search can begin at 0, or a bacward search can begin at length-1.This function will replace findTile, which will be deprecated.
+	 *
+	 * @param startPos - Position at which to start the search
+	 * @param clientId - clientId dictating the perspective to search from
+	 * @param tileLabel - Label of the tile to search for
+	 * @param tilePrecedesPos - Whether the desired tile comes before (true) or after (false) `startPos`
+	 */
+	public walkToFindTile(
+		startPos: number,
+		clientId: number,
+		tileLabel: string,
+		tilePrecedesPos = true,
+	) {
+		const searchInfo: IReferenceSearchInfo = {
+			mergeTree: this,
+			tilePrecedesPos,
+			tileLabel,
+		};
+
+		const tileShiftHelper = (node: IMergeNode) => {
+			tileShift(node, 0, UniversalSequenceNumber, clientId, startPos, -1, searchInfo);
+		};
+
+		const recordTileStartHelper = (seg: ISegment): boolean | undefined => {
+			const result: boolean = recordTileStart(
+				seg,
+				0,
+				UniversalSequenceNumber,
+				clientId,
+				startPos,
+				-1,
+				searchInfo,
+			);
+			if (!Marker.is(seg)) {
+				return true;
+			}
+			return result;
+		};
+		const { segment } = this.getContainingSegment(startPos, UniversalSequenceNumber, clientId);
+		if (segment === undefined) {
+			return undefined;
+		}
+
+		depthFirstNodeWalk(
+			this.root,
+			this.root.children[segment.index],
```

Comment:
rather than root, should these be segment.parent? most roots will not have segments as children.

Reply:
add a test case with lots of segments `Array.from({length: MaxNodesInBlock * 2).forEach((_,i)=>client.insertText(0,i.toString()))`

Reply:
> should these be segment.parent

This change makes sense to me, but CI now fails since parent is deprecated and searchForTile is public.